src/sentry/snuba/sessions.py

```diff
@@ -916,3 +944,89 @@ def __get_scope_value_for_release(
     elif scope == "crash_free_users":
         scope_value = rq_row["users_crashed"] / rq_row["users"]
     return scope_value
+
+
+def __get_crash_free_rate(project_ids, start, end, rollup, projects_crash_free_rate_dict, key):
+    """
+    Helper function that executes a snuba query on project_ids to fetch the number of crashed
+    sessions and total sessions and returns the crash free rate for those project_ids.
+    Inputs:
+        * project_ids
+        * start
+        * end
+        * rollup
+        * projects_crash_free_rate_dict: Dictionary that contains project ids and their
+            corresponding crash free rates
+        * key: represents the key name that this crashFreeRate represents either
+            `currentCrashFreeRate` or `previousCrashFreeRate`
```

Comment:
Its a bit weird that these are being passed in from the outside, even though it reduces duplication in the caller.
I think its always a good idea to just return data instead of mutating an outparam. The fn should not have `get` in its name then, as it does not return anything. Rather `_save_crash_free_rate_into` or something, but meh.